src/main/java/org/osc/manager/rest/server/api/DeviceApis.java

```diff
@@ -0,0 +1,391 @@
+import org.osc.manager.ism.api.IsmDeviceApi;
+import org.osc.manager.ism.entities.DeviceEntity;
+import org.osc.manager.ism.entities.DeviceMemberEntity;
+import org.osc.manager.rest.server.SecurityManagerServerRestConstants;
+import org.osc.sdk.manager.element.ApplianceSoftwareVersionElement;
+import org.osc.sdk.manager.element.DistributedApplianceElement;
+import org.osc.sdk.manager.element.DomainElement;
+import org.osc.sdk.manager.element.ManagerDeviceElement;
+import org.osc.sdk.manager.element.VirtualSystemElement;
+import org.osc.sdk.manager.element.VirtualizationConnectorElement;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.transaction.control.TransactionControl;
+
+@Component(service = DeviceApis.class)
+@Path(SecurityManagerServerRestConstants.SERVER_API_PATH_PREFIX + "/devices")
+@Consumes({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+@Produces({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+public class DeviceApis {
+
+    private static final Logger logger = Logger.getLogger(DeviceApis.class);
+    private EntityManager em;
+    private TransactionControl txControl;
+    private IsmDeviceApi api;
+
+    public static final class VSElement implements VirtualSystemElement {
+
+        private final Long id;
+
+        private final String name;
+
+        public VSElement(Long id, String name) {
+            this.id = id;
+            this.name = name;
+        }
+
+        @Override
+        public VirtualizationConnectorElement getVirtualizationConnector() {
+            return null;
+        }
+
+        @Override
+        public String getName() {
+            return this.name;
+        }
+
+        @Override
+        public String getMgrId() {
+            return null;
+        }
+
+        @Override
+        public byte[] getKeyStore() {
+            return null;
+        }
+
+        @Override
+        public Long getId() {
+            return this.id;
+        }
+
+        @Override
+        public DomainElement getDomain() {
+            return null;
+        }
+
+        @Override
+        public DistributedApplianceElement getDistributedAppliance() {
+            return null;
+        }
+
+        @Override
+        public ApplianceSoftwareVersionElement getApplianceSoftwareVersion() {
+            return null;
+        }
+    }
+
+    /**
+     * init the DB
+     *
+     *
+     */
+    public void init(EntityManager em, TransactionControl txControl) throws Exception {
+        this.em = em;
+        this.txControl = txControl;
+    }
+
+    /**
+     * Creates the Device for a given VS
+     *
+     * @return Device
+     * @throws Exception
+     */
+    @POST
+    public String createDevice(DeviceEntity entity) throws Exception {
+
+        logger.info("Creating Device Entity with Name:...." + entity.getName());
+        logger.info("and with VS ID:...." + entity.getVsId());
+
+        VSElement vsElement = new VSElement(entity.getVsId(), entity.getName());
```

Comment:
I believe you are doing this so you can reuse the code in the IsmDeviceApi but it is a bit odd. Instead you should create a public method in the IsmDeviceApi that calls the db, like createDevice that takes a device entity, see my previous comment on this.